src/libraries/System.Runtime.Numerics/tests/BigInteger/BigNumberTools.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Numerics;
+using System.Reflection;
+
+namespace BigNumberTools
+{
+    public class Utils
+    {
+        private static TypeInfo InternalNumber
+        {
+            get
+            {
+                if (s_lazyInternalNumber == null)
+                {
+                    Type t = typeof(BigInteger).Assembly.GetType("System.Numerics.BigNumber");
+                    if (t != null)
+                    {
+                        s_lazyInternalNumber = t.GetTypeInfo();
+                    }
+                }
+                return s_lazyInternalNumber;
+            }
+        }
+
+        private static volatile TypeInfo s_lazyInternalNumber;
```

Comment:
Can you add a comment, which explains why this is volatile.

Reply:
This commits explains the reason. Actually, I didn't understand why volatile is required.
https://github.com/dotnet/corefx/pull/18844/commits/8918578c492f979d639e633107825f00793c3d05